span-processing-config-service-impl/src/main/java/org/hypertrace/span/processing/config/service/apinamingrules/DefaultApiNamingRulesManager.java

```diff
@@ -0,0 +1,269 @@
+            .filter(
+                apiNamingRuleInfo ->
+                    apiNamingRuleInfo.getRuleConfig().hasSegmentMatchingBasedConfig())
+            .map(apiNamingRuleInfo -> buildApiNamingRule(requestContext, apiNamingRuleInfo))
+            .collect(Collectors.toUnmodifiableList());
+
+    List<ApiNamingRule> apiSpecBasedRules =
+        request.getRulesInfoList().stream()
+            .filter(apiNamingRuleInfo -> apiNamingRuleInfo.getRuleConfig().hasApiSpecBasedConfig())
+            .map(apiNamingRuleInfo -> buildApiNamingRule(requestContext, apiNamingRuleInfo))
+            .collect(Collectors.toUnmodifiableList());
+
+    return buildApiNamingRuleDetails(
+        this.apiNamingRulesConfigStore.upsertObjects(
+            requestContext,
+            Stream.concat(segmentMatchingBasedRules.stream(), apiSpecBasedRules.stream())
+                .collect(Collectors.toUnmodifiableList())));
+  }
+
+  @Override
+  public ApiNamingRuleDetails updateApiNamingRule(
+      RequestContext requestContext, UpdateApiNamingRuleRequest request) throws StatusException {
+    UpdateApiNamingRule updateApiNamingRule = request.getRule();
+    ApiNamingRule existingRule =
+        this.apiNamingRulesConfigStore
+            .getData(requestContext, updateApiNamingRule.getId())
+            .orElseThrow(Status.NOT_FOUND::asException);
+    ApiNamingRule updatedRule = buildUpdatedRule(existingRule, updateApiNamingRule);
+
+    return buildApiNamingRuleDetails(
+        this.apiNamingRulesConfigStore.upsertObject(requestContext, updatedRule));
+  }
+
+  @Override
+  public List<ApiNamingRuleDetails> updateApiNamingRules(
+      RequestContext requestContext, UpdateApiNamingRulesRequest request) {
+
+    Map<String, UpdateApiNamingRule> apiNamingRuleMap =
+        request.getRulesList().stream()
+            .collect(Collectors.toUnmodifiableMap(UpdateApiNamingRule::getId, Function.identity()));
+
+    List<ApiNamingRule> existingRules =
+        apiNamingRulesConfigStore.getAllData(requestContext).stream()
+            .map(ApiNamingRuleDetails::getRule)
+            .filter(apiNamingRule -> apiNamingRuleMap.containsKey(apiNamingRule.getId()))
+            .collect(Collectors.toUnmodifiableList());
+
+    List<ApiNamingRule> updatedRules = new ArrayList<>();
+    for (ApiNamingRule existingRule : existingRules) {
+      updatedRules.add(buildUpdatedRule(existingRule, apiNamingRuleMap.get(existingRule.getId())));
+    }
+    return buildApiNamingRuleDetails(
+        this.apiNamingRulesConfigStore.upsertObjects(requestContext, updatedRules));
+  }
+
+  @Override
+  public void deleteApiNamingRule(
+      RequestContext requestContext, DeleteApiNamingRuleRequest request) {
+    // TODO: need to handle priorities
+    this.apiNamingRulesConfigStore
+        .deleteObject(requestContext, request.getId())
+        .orElseThrow(Status.NOT_FOUND::asRuntimeException);
+  }
+
+  @Override
+  public void deleteApiNamingRules(
```

Comment:
will it work? I remember the id gets map to context value in the config object. If you need to expose these api you should add this as attribute in the api naming rule itself

Reply:
We are using `IdentifiedObjectStore` for api naming rules. The delete api of this object store just takes the id alone.